src/electron/platform/android/scan-controller.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { TelemetryEventHandler } from 'background/telemetry/telemetry-event-handler';
+import { InstanceCount } from 'common/extension-telemetry-events';
+import { SCAN_COMPLETED, SCAN_FAILED } from 'electron/common/electron-telemetry-events';
+import { PortPayload } from 'electron/flux/action/device-action-payloads';
+import { ScanActions } from 'electron/flux/action/scan-actions';
+import { FetchScanResultsType } from 'electron/platform/android/fetch-scan-results';
+import { RuleResultsData, ScanResults } from 'electron/platform/android/scan-results';
+
+export class ScanController {
+    constructor(
+        private readonly scanActions: ScanActions,
+        private readonly fetchScanResults: FetchScanResultsType,
+        private readonly telemetryEventHandler: TelemetryEventHandler,
+        private readonly getCurrentDate: () => Date,
+    ) {}
+
+    public initialize(): void {
+        this.scanActions.scanStarted.addListener(this.onScanStarted);
+    }
+
+    private onScanStarted = (payload: PortPayload) => {
+        const port = payload.port;
+
+        const scanStartedTime = this.getCurrentDate().getTime();
+
+        this.fetchScanResults(port)
```

Comment:
how about logging telemetry event on scan started here, so that all telemetry events related to scan is here

Reply:
sounds good

Reply:
done